engine/src/main/java/org/destinationsol/assets/audio/OggMusicManager.java

```diff
@@ -133,4 +141,70 @@ private void stopMusic() {
     public void changeVolume(GameOptions options) {
         currentlyPlaying.setVolume(options.musicVolume.getVolume());
     }
+
+    /**
+     * Registers all music from the module of shipName
+     *
+     * @param shipName Name of the ship that is loaded when creating a new game
+     */
+    public void registerModuleMusic(String shipName) {
+        Set<ResourceUrn> configUrnList = Assets.getAssetHelper().list(Json.class, "[a-zA-Z]*:playerSpawnConfig");
+
+        for (ResourceUrn configUrn : configUrnList) {
+            Json json = Assets.getJson(configUrn.toString());
+            JSONObject rootNode = json.getJsonValue();
+
+            if (rootNode.keySet().contains(shipName) && rootNode.get(shipName) instanceof JSONObject) {
+                String moduleName = configUrn.toString().split(":")[0];
+                Json musicJson;
+
+                try {
+                    musicJson = Assets.getJson(moduleName + ":musicConfig");
+                } catch (RuntimeException e) {
+                    logger.warn("Music config not found for module " + moduleName + ":musicConfig");
+                    return;
+                }
+
+                MusicConfig musicConfig = MusicConfig.load(moduleName, musicJson.getJsonValue());
+                Map<String, List<String>> musicMap = musicConfig.getMusicMap();
+
+                this.musicMap.remove(MENU_MUSIC_SET);
+                for (String music : musicMap.get(MENU_MUSIC_SET)) {
+                    registerMusic(MENU_MUSIC_SET, music);
+                }
+                for (String music : musicMap.get(GAME_MUSIC_SET)) {
+                    registerMusic(GAME_MUSIC_SET, music);
+                    logger.info("Registered " + music);
+                }
+            }
+
+            json.dispose();
+        }
+
+
+        registerMusic(GAME_MUSIC_SET, "engine:cimmerianDawn");
+        registerMusic(GAME_MUSIC_SET, "engine:intoTheDark");
+        registerMusic(GAME_MUSIC_SET, "engine:spaceTheatre");
+
+    }
+
+    /**
+     * Registers all module menu music
+     */
+    public void registerAllMenuMusic() {
+        Set<ResourceUrn> configUrnList = Assets.getAssetHelper().list(Json.class, "[a-zA-Z]*:musicConfig");
+
+        for (ResourceUrn configUrn : configUrnList) {
+            String urnString = configUrn.toString();
+            Json musicJson = Assets.getJson(urnString);
+            JSONObject musicNode = musicJson.getJsonValue();
+
+            for (Object musicFileName : musicNode.getJSONArray("menuMusic")) {
+                if (!(musicFileName instanceof String))
```

Comment:
We follow 1TBS exclusively https://github.com/MovingBlocks/Terasology/wiki/Code-Conventions